code/modules/projectiles/projectile/hook.dm

```diff
@@ -0,0 +1,192 @@
+		firer.Beam(src,icon_state=beam_state,icon='icons/effects/beam.dmi',time=60, maxdistance=10,beam_type=/obj/effect/ebeam,beam_sleep_time=1)
+
+	if(launcher_intent)
+		switch(launcher_intent)
+			if(I_HURT)
+				check_armour = "bullet"
+				damage *= 3
+				sharp = 1
+				agony = 20
+			if(I_GRAB)
+				check_armour = "melee"
+				damage_type = HALLOSS
+			if(I_DISARM)
+				check_armour = "melee"
+				if(prob(30))	// A chance for a successful hit to either knock someone down, or cause minor disorientation.
+					weaken = 1
+				else
+					stun = 2
+					eyeblur = 3
+			if(I_HELP)
+				silenced = 1
+				damage_type = HALLOSS
+
+	..() // Does the regular launching stuff.
+
+/obj/item/projectile/energy/hook/on_hit(var/atom/target, var/blocked = 0, var/def_zone = null)
+	if(..())
+		perform_intent_unique(target)
+
+/obj/item/projectile/energy/hook/on_impact(var/atom/A)
+	perform_intent_unique(get_turf(A))
+
+/obj/item/projectile/energy/hook/proc/ranged_disarm(var/mob/living/carbon/human/H)
+	if(istype(H))
+		var/list/holding = list(H.get_active_hand() = 60, H.get_inactive_hand() = 40)
+
+		for(var/obj/item/weapon/gun/W in holding)	// Guns are complex devices, both of a mechanical and electronic nature. A weird gravity ball or other type of object trying to pull or grab it is likely not safe.
+			if(W && prob(holding[W]))
+				var/list/turfs = list()
+				for(var/turf/T in view())
+					turfs += T
+				if(turfs.len)
+					var/turf/target = pick(turfs)
+					visible_message("<span class='danger'>[H]'s [W] goes off due to \the [src]!</span>")
+					return W.afterattack(target,H)
+
+		if(!(H.species.flags & NO_SLIP) && prob(50))
+			var/armor_check = H.run_armor_check(def_zone, "melee")
+			H.apply_effect(3, WEAKEN, armor_check)
+			playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+			if(armor_check < 60)
+				visible_message("<span class='danger'>\The [src] has pushed [H]!</span>")
+			else
+				visible_message("<span class='warning'>\The [src] attempted to push [H]!</span>")
+			return
+
+		else
+			if(H.break_all_grabs(firer))
+				playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+				return
+
+			for(var/obj/item/I in holding)
+				if(I)
+					H.drop_from_inventory(I)
+					visible_message("<span class='danger'>\The [src] has disarmed [H]!</span>")
+					playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+					return
+
+
+/obj/item/projectile/energy/hook/proc/perform_intent_unique(atom/target)
+	playsound(src.loc, impact_sound, 40, 1)
+	var/success = FALSE
+	if(istype(target,/turf))
+		if(launcher_intent)
+			if(launcher_intent != I_HELP && !done_mob_unique)
+				var/target_mob = pick(/mob/living in target.contents)
+
+				if(!target_mob)
+					return
+
+				if(Bump(target_mob, forced=1))	//If we hit a turf, try to force an interaction with a mob on the turf.
+					done_mob_unique = TRUE
+				success = TRUE
+			else if(firer)
+				var/obj/T
+
+				if(original in target.contents && istype(original, /obj))
```

Comment:
Original doesn't appear to be defined anywhere, so I'm not sure how this compiled.

Reply:
Original is the projectile's 'original' target, the thing the player clicked on to fire the gun.